### ISSUE TITLE
Homebridge on Raspberry Pi stops working after a few hours

### ISSUE THREAD

**Issue by danwalton84** [nfarina/homebridge#476]
Hi, I'm not sure exactly how long it takes, but my Homebridge install on a Raspberry Pi 2 works perfectly but then stops working after a time (eg overnight it won't work in the morning). Siri can't see the relevant devices (Nest and Harmony Hub).

This occurs whether connected to my Apple Timecapsule via Ethernet or via a wifi dongle.

Have I missed something during setup to keep Homebridge running permanently? Or is the Raspberry Pi going to sleep or closing the Terminal session?

Thanks,

**Comment by danwalton84** [comment]
Meant to say I'm running Rasbian from Noobs.

**Comment by danwalton84** [comment]
Just a bit of additional information: I plugged in the Rasberry Pi to my TV, and Terminal was still on the screen but the last entry was a socket hang up error.

Any ideas?

**Comment by cflurin** [comment]
Disabling the WiFi power management on the RPI might help. Example for the Realtek 8192CU chipset:
```
echo "options 8192cu rtw_power_mgnt=0 rtw_enusbss=0" | sudo tee --append /etc/modprobe.d/8192cu.conf
```

**Comment by danwalton84** [comment]
Power management is off, but it still stopped working overnight. This morning was an Econnreset error as the last entry in the Terminal window.

**Comment by KhaosT** [comment]
@danwalton84 ECONNRESET is a different error. Can you post the terminal output when that error happens?

**Comment by danwalton84** [comment]
I've just restarted it, but next time it happens I will. What do I need to do? Just give you the last error message I get?

**Comment by KhaosT** [comment]
Yeah, just copy and paste the error message (and maybe stack trace if there is) here :)

**Comment by xalphay** [comment]
[5/18/2016, 9:25:04 PM] Homebridge is running on port 51826.
error at req:  socket hang up
error at req:  socket hang up
error at req:  socket hang up
error at req:  connect ETIMEDOUT 107.20.201.51:443
^C[5/18/2016, 10:42:54 PM] Got SIGINT, shutting down Homebridge...

I  got the same issue, I will try to Disable the WiFi power management, and then I will be back to give the feedback.

**Comment by xalphay** [comment]
Maybe I join the wrong Router...
`  EventedHTTPServer [::ffff:192.168.1.159] Sending HTTP event 'keepalive' with data: {"characteristics":[]} +5m
  EventedHTTPServer [::ffff:192.168.1.83] Sending HTTP event 'keepalive' with data: {"characteristics":[]} +3ms
error at req:  socket hang up
error at req:  connect ETIMEDOUT 107.20.201.51:443
  EventedHTTPServer [::ffff:192.168.1.159] Client connection closed +10m
  EventedHTTPServer [::ffff:192.168.1.159] HTTP connection was closed +2ms
  EventedHTTPServer [::ffff:192.168.1.159] HTTP server was closed +1ms
  EventedHTTPServer [::ffff:192.168.1.159] New connection from client +658ms
`

After the error line,  a new connection occured, I will keep an eye on it.

**Comment by rcaslis** [comment]
I'm seeing exactly the same behavior on my Raspberry Pi 3.  I'm running the latest homebridge with the alarmdotcom plugin.

**Comment by cflurin** [comment]
@rcaslis: are you using a Wifi dongle?

**Comment by rcaslis** [comment]
Not exactly.   I was using the built-in wifi on the Pi 3.  I had the same problem with power management on & off. 

I've switched to a wired connection and things seem stable now. 


- Russ

**Comment by cflurin** [comment]
Try adding this line to `/etc/rc.local`
```
sudo iwconfig wlan0 power off
```
This will keep power management off after reboot.

**Comment by rcaslis** [comment]
Well, it's happening again now even while running on a wired connection with wireless powered down.  Not sure what's going on.

**Comment by exsilium** [comment]
Hi! I've recently started testing Homebridge for stability using Raspberry Pi 2 and Ubuntu Xenial Xerus release and up to now it seems to be running very solid (I'm using wired connection). Whatever you are encountering might be Raspbian specific. If you have a spare microSD card, I'd recommend you to give Ubuntu a go to see if you can reproduce the issue or not. Here's more info about the image and the installation: https://wiki.ubuntu.com/ARM/RaspberryPi (includes links for official Pi2 and unofficial Pi3 images).

**Comment by cflurin** [comment]
I'm running homebridge on rpi2 and rpi zero (both with a wifi dongle) for months without any issues!

**Comment by thoward1234** [comment]
I found the Harmony Hub plugin reduces reliability for HomeBridge.  I used a different setup (Harmony->Smartthings->Homebridge) and have not had an issue since

**Comment by jer78** [comment]
The HarmonyHub 0.2.0-alpha fixed the crashing for me.

**Comment by peaz** [comment]
I was having the same problem. But after a while I figured out why that's the case for my install. I've blogged it [here](https://www.atpeaz.com/fixing-homebridge-no-accessories-responding-problem/) so I would remind myself of it for future installs, but long story short
1. Check and make sure multicast is enabled on your router. Different routers may have different settings that need to be configured to get this done properly. I'm using an Asus AC87U and there were a few settings that I needed to ensure is configured in the IPTV settings.
2. Disable sleep mode for the WiFi module on the Raspberry Pi.. yup. Took me a while before I figured this out.

Hopefully these would help you too. My install has been stable ever since.
